cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,1128 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#if defined(_CCCL_CUDA_COMPILER) && _CCCL_CUDACC_VER < 1105000
+_CCCL_NV_DIAG_SUPPRESS(186)
+#  include <cuda_pipeline_primitives.h>
+// we cannot re-enable the warning here, because it is triggered outside the translation unit
+// see also: https://godbolt.org/z/1x8b4hn3G
+#endif // defined(_CCCL_CUDA_COMPILER) && _CCCL_CUDACC_VER < 1105000
+
+#include <cub/detail/uninitialized_copy.cuh>
+#include <cub/device/device_for.cuh>
+#include <cub/util_arch.cuh>
+#include <cub/util_type.cuh>
+
+#include <thrust/detail/raw_reference_cast.h>
+#include <thrust/system/cuda/detail/core/triple_chevron_launch.h>
+#include <thrust/type_traits/is_contiguous_iterator.h>
+#include <thrust/type_traits/is_trivially_relocatable.h>
+
+#include <cuda/cmath>
+#include <cuda/ptx>
+#include <cuda/std/__algorithm/clamp.h>
+#include <cuda/std/__algorithm/max.h>
+#include <cuda/std/__algorithm/min.h>
+#include <cuda/std/array>
+#include <cuda/std/expected>
+#include <cuda/std/tuple>
+#include <cuda/std/type_traits>
+#include <cuda/std/utility>
+
+#include <cooperative_groups.h>
+#include <cooperative_groups/memcpy_async.h>
+
+CUB_NAMESPACE_BEGIN
+
+// the ublkcp kernel needs PTX features that are only available and understood by CTK 12 and later
+#if _CCCL_CUDACC_VER_MAJOR >= 12
+#  define _CUB_HAS_TRANSFORM_UBLKCP
```

Comment:
what about if we move this macro in a common place for all algorithms?

Reply:
This macro is more of an implementation helper for me. If I moved it to somewhere common, I would also need to define more clearly what it means. For now it means: enabled when the ublkcp transform kernel can for sure be compiled. That's not very useful outside of `cub::DeviceTransform` IMO.

Reply:
@fbusato We already have a generic set of macros to detect the nvcc version [here](https://github.com/NVIDIA/cccl/blob/main/libcudacxx/include/cuda/std/__cccl/compiler.h#L78-L96)

I agree with @bernhardmgruber that this macro is best kept localized to its uses